cli/snapshot.go

```diff
@@ -76,199 +86,231 @@ func (runOpts *runOptionsType) DumpSnapshot(ctx *cli.Context) error {
 		return errDumpTerminal
 	}
 
-	// Make sure we don't dump to rootfs
-	// which would cause the system to freeze
-	var stdoutStat unix.Stat_t
-	var rootStat unix.Stat_t
-	if err := unix.Fstat(fd, &stdoutStat); err != nil {
-		return errors.Wrap(err, "Unable to stat output file")
-	}
-	if err := unix.Stat(ctx.String("fs-path"), &rootStat); err != nil {
-		return errors.Wrap(err, "Unable to stat root filesystem")
-	}
-	if stdoutStat.Dev == rootStat.Dev {
-		return errors.New(
-			"Dumping the filesystem to itself is not permitted")
```

Comment:
Are you relying completely on the freeze timer now? This message is being tested for, so we need to change the test, in that case.

Reply:
Yes, I don't think we should treat this redirection case any different from pipes. So just let the device freeze until the timer expires. That's the punishment of incorrect usage :smiling_imp:

Reply:
Hehe ok, I will adjust the test. :-)